assets/js/base/components/tabs/index.js

```diff
@@ -1,93 +1,79 @@
 /**
  * External dependencies
  */
-import { useState, useEffect } from '@wordpress/element';
 import { withInstanceId } from '@woocommerce/base-hocs/with-instance-id';
 import classnames from 'classnames';
 import { __ } from '@wordpress/i18n';
-
+import { useTabState, Tab, TabList, TabPanel } from 'reakit/Tab';
 /**
  * Internal dependencies
  */
 import './style.scss';
-import TabButton from './tab-button';
 
 const Tabs = ( {
 	className,
 	onSelect = () => null,
 	tabs,
 	activeClass = 'is-active',
 	initialTabName,
-	instanceId,
 	ariaLabel = __( 'Tabbed Content', 'woo-gutenberg-products-block' ),
-	children,
+	instanceId,
 	id,
 } ) => {
-	const [ selected, setSelected ] = useState(
-		() => initialTabName || ( tabs.length > 0 ? tabs[ 0 ].name : '' )
-	);
-	// Maybe set selected tab if it's empty but we have tabs or an initial tab
-	// This covers instances when the selected value isn't initialized on mount.
-	useEffect( () => {
-		if ( ! selected && tabs.length > 0 ) {
-			setSelected(
-				initialTabName || ( tabs.length > 0 ? tabs[ 0 ].name : '' )
-			);
-		}
-	}, [ selected, tabs, initialTabName ] );
-	if ( ! selected ) {
+	const initialTab =
+		( initialTabName || !! tabs.length ) &&
+		`${ instanceId }-${ initialTabName ? initialTabName : tabs[ 0 ].name }`;
+	const tabState = useTabState( {
+		selectedId: initialTab,
+	} );
+	if ( tabs.length === 0 ) {
 		return null;
 	}
-	const handleClick = ( tabKey ) => {
-		setSelected( tabKey );
-		onSelect( tabKey );
-	};
-	const selectedTab = tabs.find( ( tab ) => tab.name === selected );
-	if ( ! selectedTab ) {
-		throw new Error( 'There is no available tab for the selected item' );
-	}
```

Comment:
tabs and content are returned on the same object, this felt redundant and caused us to guard against empty Tabs from `paymentMethod`
is there a scenario in which this will trigger?